### PR TITLE
fix(web): wire permission approval UI for Kandev MCP tools

### PR DESCRIPTION
**Pull request by luancm**
KandevToolMessage adapter ignored the `permissionMessage` prop, so MCP tools that block on a permission request rendered the result row but no Approve / Deny buttons — users could not unblock the agent. The permission handlers (`parsePermission`, `usePermissionResponseHandlers`) are now extracted into `use-permission-handlers.ts` and reused from both `ToolCallMessage` and `KandevToolMessage`, and `KandevRow` shows an amber pending clock icon plus hides the result-count summary while the prompt is open.

## Important Changes

- `KandevToolMessage` now accepts `permissionMessage`, renders `PermissionActionRow` when pending, and the dispatcher in `message-renderer.tsx` looks up the permission by `tool_call_id` (same shape as the generic adapter).
- `KandevStatusIcon` adds a `pending` branch (amber `IconClock`); `KandevRow` suppresses the inline summary while `status === "pending"` so users do not see a misleading "0 workspaces" before the result lands.
- New mock-agent scenario `kandev-mcp-permission` emits `mcp__kandev__list_workspaces_kandev` + a permission request for deterministic e2e coverage.

## Validation

- `pnpm --filter @kandev/web typecheck`
- `pnpm --filter @kandev/web lint`
- `pnpm --filter @kandev/web test components/task/chat/messages/` (44/44)
- `make -C apps/backend test build lint`
- TDD-verified the two UX unit tests: stashing the `shared.tsx` changes reproduces 2 failures (`tabler-icon-clock` missing, `"0 workspaces"` present); restoring the fix returns 16/16.
- New Playwright spec `Kandev custom MCP tools render permission approval UI` reproduces the original bug end-to-end.

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.

### CODE REVIEW

File: apps/web/components/task/chat/messages/kandev-tool-message.tsx [hunk 1]
```diff
@@ -54,9 +57,15 @@ export function hasKandevRenderer(comment: Message): boolean {
...
   comment,
+  permissionMessage,
 }: KandevToolMessageProps) {
   const meta = comment.metadata as ToolCallMetadata | undefined;
   const renderer = getKandevRenderer(kandevStemOf(comment));
+  const { permissionMetadata, isPermissionPending } = parsePermission(permissionMessage);
+  const { isResponding, handleApprove, handleReject } = usePermissionResponseHandlers({
+    permissionMetadata,
+    permissionMessage,
+  });
```
[the remark this answers is not in the document]
Comment (luancm) [reply apps/web/components/task/chat/messages/kandev-tool-message.tsx]: Fixed in 6ec66de9 — added `resolveKandevStatus(meta?.status, permissionStatus)` so the row drops the amber clock and exposes the result summary as soon as the permission_request is stamped `approved`/`rejected`, without waiting for the next tool_call update. Unit test `drops the amber pending clock once the permission resolves to approved` covers the lag window.

File: apps/web/e2e/tests/chat/permission-approval.spec.ts [hunk 2]
```diff
@@ -121,4 +121,37 @@ test.describe("Permission approval persistence", () => {
...
+    await testPage.goto(`/t/${task.id}`);
+
+    const session = new SessionPage(testPage);
+    await session.waitForLoad();
+
+    // The mock agent will trigger mcp__kandev__list_workspaces_kandev and block.
+    // If the custom Kandev renderer correctly wired permissions, we should see 1 button.
+    await expect(session.permissionApproveButtons()).toHaveCount(1, { timeout: 30_000 });
+    await session.permissionApproveButtons().first().click();
+
```
[the remark this answers is not in the document]
Comment (luancm) [reply apps/web/e2e/tests/chat/permission-approval.spec.ts]: Fixed in 6ec66de9 — dropped the redundant `.first()`.

File: apps/web/components/task/chat/messages/kandev-tool-message.tsx [hunk 3]
```diff
@@ -5,9 +5,28 @@ import type { Message } from "@/lib/types/http";
...
+// resolveKandevStatus collapses the tool-call status and the permission
+// decision into the single status the renderer should display. The tool_call
+// stays "pending" until the next backend update, so without this the row
+// keeps showing the amber clock after approve/reject — the resolved decision
+// wins as soon as the permission_request is stamped.
+function resolveKandevStatus(
+  toolStatus: ToolCallMetadata["status"],
+  permissionStatus: PermissionRequestMetadata["status"],
+): KandevStatus {
+  if (permissionStatus === "approved") return "complete";
```
Comment (zeval) [inline apps/web/components/task/chat/messages/kandev-tool-message.tsx]: Blocker: this maps an approved permission decision to a completed tool status unconditionally. Because the approved permission_request remains paired with the tool_call, the Kandev row can render as complete even while the actual tool_call is still pending/running, and it can also mask a later tool_call error. It can briefly show result summaries from an empty/missing result as if the tool succeeded.

Please keep permission approval separate from tool completion: rejection can force an error state, but approval should not override the tool_call status. If the goal is only to remove the amber pending-permission clock after approval, model that as a separate display state while still passing through the real tool status.

**Comment by zeval** [comment]
Thanks! Left some comments 👍

File: apps/web/components/task/chat/messages/kandev-tool-message.tsx (hunk 3 above)
Comment (luancm) [reply apps/web/components/task/chat/messages/kandev-tool-message.tsx]: Good catch — addressed in ea821763. The status override is gone; the row now reads the real `tool_call.status` always, and the permission overlay is plumbed separately via a new `KandevPermissionUIContext` (`"pending" | "rejected" | undefined`).

- **Approved** falls through to `undefined`, so an approved-but-still-pending tool stays pending, and a later `tool_call` error still renders the red X (regression tests `does NOT mark the tool complete just because the permission was approved` and `still renders an error when the tool_call errors after an approved permission`).
- **Pending** drives only the waiting-on-user affordances: amber clock + hidden summary.
- **Rejected** layers a red X overlay on the header that wins over the tool's own success state, as you suggested rejection may.

Empty-result summaries no longer surface during the approve→start race window because the underlying status is still `pending` and the renderer's "0 workspaces" caption stays suppressed by the same permission overlay until the real result lands.
